vcs/gitcmd/repo.go

```diff
@@ -962,6 +962,31 @@ func (r *Repository) Committers(opt vcs.CommittersOptions) ([]*vcs.Committer, er
 	return committers, nil
 }
 
+func (r *Repository) ListFiles(at vcs.CommitID) ([]string, error) {
+	r.editLock.RLock()
+	defer r.editLock.RUnlock()
+
+	if at == "" {
+		at = "HEAD"
+	}
+
+	cmd := exec.Command("git", "ls-files", "--with-tree", string(at))
```

Comment:
> Help me understand why using `-z` flag and splitting on `byte(0)` character is preferable. Is it performance or stability (or both)?

The only difference is in the output:

> Unless the -z option is used, TAB, LF, and backslash characters in pathnames are represented as \t, \n, and \\, respectively.

So it's a matter if we prefer tabs, backslashes or LF characters to be present in the output as is, or converted to their escaped forms.